code/__HELPERS/math.dm

```diff
@@ -0,0 +1,107 @@
+/**
+ * Get a list of turfs in a line from `starting_atom` to `ending_atom`.
+ *
+ * Uses the ultra-fast [Bresenham Line-Drawing Algorithm](https://en.wikipedia.org/wiki/Bresenham%27s_line_algorithm).
+ */
+/proc/get_line(atom/starting_atom, atom/ending_atom)
+	var/px = starting_atom.x //starting x
+	var/py = starting_atom.y
+	var/list/line = list(locate(px, py, starting_atom.z))
+	var/dx = ending_atom.x - px //x distance
+	var/dy = ending_atom.y - py
+	var/dxabs = abs(dx)//Absolute value of x distance
+	var/dyabs = abs(dy)
+	var/sdx = SIGN(dx) //Sign of x distance (+ or -)
+	var/sdy = SIGN(dy)
+	var/x = dxabs >> 1 //Counters for steps taken, setting to distance/2
+	var/y = dyabs >> 1 //Bit-shifting makes me l33t.  It also makes get_line() unnessecarrily fast.
+	var/j //Generic integer for counting
+	if(dxabs >= dyabs) //x distance is greater than y
+		for(j = 0; j < dxabs; j++)//It'll take dxabs steps to get there
+			y += dyabs
+			if(y >= dxabs) //Every dyabs steps, step once in y direction
+				y -= dxabs
+				py += sdy
+			px += sdx //Step on in x direction
+			line += locate(px, py, starting_atom.z)//Add the turf to the list
+	else
+		for(j=0 ;j < dyabs; j++)
+			x += dxabs
+			if(x >= dyabs)
+				x -= dyabs
+				px += sdx
+			py += sdy
+			line += locate(px, py, starting_atom.z)
+	return line
+
+/proc/can_see_through(turf/from_turf, turf/to_turf)
```

Comment:
doc, this would be great as a macro but i won't force you because macros are shit

Reply:
i would do it as a macro but somewhere i need it to have a return value, would probably need to be  2 macros, plus i dont remember macro syntax lmao